src/tests/playwright/pages/util/recorder/recorder-main.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview AudioWorklet Recorder Node to record raw PCM data from
+ * Web Audio Graph.
+ */
+
+/**
+ * Creates an AudioWorklet recorder node to record input for a specified
+ * length of time (seconds). Passes audio through to output.
+ * @param {AudioContext} context - The AudioContext to create the recorder node.
+ * @param {number} recordLength - The length in seconds to record for.
+ * @return {Object} An object containing:
+ *   - {AudioWorkletNode} recorder: The recorder AudioWorkletNode.
+ *   - {Promise<AudioBuffer>} bufferPromise: A promise that resolves to an
+ *   AudioBuffer once filled.
+ */
+export const record = async (context, recordLength) => {
+  console.assert(context instanceof AudioContext);
+  console.assert(typeof recordLength === 'number' && recordLength > 0);
+
+  const maxSamples = recordLength * context.sampleRate;
+
+  await context.audioWorklet.addModule('./util/recorder/recorder-processor.js');
+
+  const recorder = new AudioWorkletNode(context, 'recorder', {
+    processorOptions: {
+      maxSamples,
+    },
+  });
+
+  let bufferResolve;
```

Comment:
This is the "resolve" function for promise, so perhaps it's better with `resolveWithTestResult()` or something similar?